### ISSUE TITLE
tagaffinityrecommender tag store not found

### ISSUE THREAD

**Issue by boogardgodig** [SeldonIO/seldon-server#16]
I'm trying tagaffinity recommender. the model looks fine. but i can't get any recommendations.
I get this error in api logs
`2016-05-11 13:00:02,173 DEBUG http-nio-8080-exec-10 UserTagAffinityRecommender [client] [test123] [] - Failed to get tag store for client client
2016-05-11 13:00:02,173 DEBUG http-nio-8080-exec-10 RecommendationPeer [client] [test123] [] - Recommender userTagAffinityRecommender returned 0 results 
`
what's tag store? why seldon can't find it?

In case it's related, this is my recommender configuration. 
/all_clients/client/algs:
`{"algorithms":
[{"name":"userTagAffinityRecommender","filters":[],"includers":["recentItemsIncluder"],
"config":[{"name":"io.seldon.algorithm.tags.attrid","value":7},
{"name":"io.seldon.algorithm.tags.useitemdim","value":"false"},
{"name":"io.seldon.algorithm.inclusion.itemsperincluder","value":1000},
{"name":"io.seldon.algorithm.clusters.minnumberitemsforvalidclusterresult","value":10},
{"name":"io.seldon.algorithm.clusters.decayratesecs","value":10800},
{"name":"io.seldon.algorithm.clusters.categorydimensionname","value":"category"}]},
{"name":"recentItemsRecommender","filters":[],"includers":[],"config":[]}],
"combiner":"firstSuccessfulCombiner"}`

and I got tags.attrid from item_attr table in seldon db
`| attr_id | name | type    | item_type | semantic |`
`|       7 | tags | VARCHAR |         1 |     NULL |`

**Comment by cliveseldon** [comment]
It looks like no model for that client has been loaded. If you ran the Spark job then check two locations in Zookeeper:

 /config/tagaffinity should have your client name e.g., client
/all_clients/<client_name>/tagaffinity should have the location of the model, e.g.
/all_clients/client/tagaffinity

**Comment by boogardgodig** [comment]
/config/tagaffinity was empty
what's tag map?

**Comment by cliveseldon** [comment]
It means there was no tag relationships for this user. Were they in the training data?

**Comment by boogardgodig** [comment]
all my items have at least one tag. and that user has actions in logs but I constantly get that error

**Comment by cliveseldon** [comment]
I suggest you look at the actual model files. They are quite simple text files. You can check which users have been loaded.

The model has settings that might mean the user has not enough activity for a tag for it to be included, in particular:
   minActionsPerUser : Int = 10,
    minTagCount : Int = 4,
 
I suggest you change these to lower numbers if you wish.

**Comment by boogardgodig** [comment]
I looked into models .when I set minActionsPerUser=1 and minTagCount=1 , minPcIncrease=0.01 with days=4 (number of days to input actions) it shows 4 tags for the user. but when i tryied other number of days the model file was empty. I can't exactly understand why using more actions doesn't genererate more tag entries.

I get 2 warnings when generating the model

`16/05/19 13:47:51 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java class`
`16/05/19 13:48:08 WARN ExponentialBackoffRetry: maxRetries too large (100). Pinning to 29
ses where applicable`

and some mapred class deprecated logs 

Eventhough it says `tagCounts size is 2051` after decpressing actions it generates an empty file

Can't I set tagaffinity so that at least k tags are selected per user?
